pkg/virt-operator/resource/generate/components/validations_generated.go

```diff
@@ -794,37 +816,75 @@ var CRDsValidation map[string]string = map[string]string{
               format: int32
               type: integer
             migrations:
-              description: MigrationConfiguration holds migration options
+              description: MigrationConfiguration holds migration options. Can be
+                overriden for specific groups of VMs though migration policies. Visit
+                https://kubevirt.io/user-guide/operations/migration_policies/ for
+                more information.
               properties:
                 allowAutoConverge:
+                  description: AllowAutoConverge allows the platform to compromise
+                    performance/availability of VMIs to guarantee successful VMI live
+                    migrations. Defaults to false
                   type: boolean
                 allowPostCopy:
+                  description: AllowPostCopy enables post-copy live migrations. Such
+                    migrations allow even the busiest VMIs to successfully live-migrate.
+                    However, events like a network failure can cause a VMI crash.
+                    If set to true, migrations will still start in pre-copy, but switch
+                    to post-copy when CompletionTimeoutPerGiB triggers. Defaults to
+                    false
                   type: boolean
                 bandwidthPerMigration:
                   anyOf:
                   - type: integer
                   - type: string
+                  description: BandwidthPerMigration limits the amount of network
+                    bandwith live migrations are allowed to use. The value is in quantity
+                    per second. Defaults to 0 (no limit)
                   pattern: ^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$
                   x-kubernetes-int-or-string: true
                 completionTimeoutPerGiB:
+                  description: CompletionTimeoutPerGiB is the maximum number of seconds
+                    1GiB can take to live-migrate. If, during any CompletionTimeoutPerGiB
+                    period, less than 1GiB is transfered from the source to the target,
+                    the migration is cancelled, unless AllowPostCopy is true. Defaults
+                    to 800
```

Comment:
Overall, I think this is accurate, but it reads as if we have a chance here of falling into a local minimum.
In reality, kubevirt will abort the migration or switch to post copy only if the migration has been running for too long, longer than the `acceptableCompletionTime`.
This `acceptableCompletionTime` is being determined by an estimated "size" of the VMI: sizeof(memory + disks) * CompletionTimeoutPerGiB; 
`CompletionTimeoutPerGiB` is more like: "how much time do we think is acceptable to transfer 1GiB"

So the migration will not be canceled if we transferred less than 1GiB in completionTimeoutPerGiB, but only when acceptableCompletionTime exceeded.

Reply:
Nice catch, I did not realize that, having incorrect documentation would actually be worse than no documentation, so thank you!